src/esp/sim/Simulator.h

```diff
@@ -99,8 +99,10 @@ class Simulator {
    * --headless mode on linux
    */
   int gpuDevice() const {
-    CORRADE_ASSERT(context_ != nullptr,
-                   "Simulator::gpuDevice: no OpenGL context.", 0);
+    if (!context_) {  // todo: be more strict and verbose here. test
```

Comment:
Let's resolve this todo. Simulator has access to the createRenderer flag, so it should know for sure whether we expect to have a context here. We should still be able to do some kind of assert.

Reply:
Pushed something to address this; let me know what you think!